server/src/main/java/org/opensearch/action/search/TransportSearchAction.java

```diff
@@ -1216,6 +1213,8 @@ private List<SearchRequestOperationsListener> createSearchListenerList(SearchReq
             searchListenersList.add(timeProvider);
         }
 
+        searchListenersList.add(new SearchRequestSlowLog(clusterService));
```

Comment:
`SearchRequestSlowLog` currently stores request phase duration so cannot be node level. We could track this elsewhere but may introduce other complexity. 
```
Map<String, Long> phaseTookMap
```

Reply:
Can we come up with approach on how this will look at node level? Just the abstractions should suffice.

Reply:
For node level to work, we need someplace to store `absoluteStartNanos` and `phaseTookMap` per request. There is `SearchContext` at shard-level and `SearchPhaseContext` at phase-level, but no such request-level context. We can do one of the following:
1. Add to `AbstractSearchAsyncAction`. This is the simplest option because this is where our SearchRequestOperationsListener is invoked on the coordinator. AbstractSearchAsyncAction holds other request-level state such as currentPhase, releasables, shardFailures, pendingExecutionsPerNode, etc...
2. Create request-level search context. For our purposes this may be overkill.
```
class SearchRequestContext() {
    private long absoluteStartNanos;
    private final Map<String, Long> phaseTookMap
}
```
3. Create a large hash map <request_id, data> inside `SearchRequestSlowLog` for all requests. As we discussed this has the potential for memory leak when requests are not properly removed so I would not recommend.

Reply:
I have gone with option 2 since we can also use `SearchRequestContext` to pass additional stats in the slow log message.

`SearchRequestSlowLog` is now created once per node instead of per request.